crypto/src/main/scala/org/bitcoins/crypto/HashType.scala

```diff
@@ -1,26 +1,29 @@
-package org.bitcoins.core.script.crypto
+package org.bitcoins.crypto
 
-import org.bitcoins.core.number.Int32
-import org.bitcoins.crypto.{ECDigitalSignature, Factory}
-import scodec.bits.ByteVector
+import scodec.bits.{ByteOrdering, ByteVector}
 
 /** Created by chris on 1/18/16.
   */
 sealed trait HashType {
-  def num: Int32
-  def byte: Byte = num.bytes.last
+  def num: Int
+  def byte: Byte = ByteVector.fromInt(num).last
 }
 
 object HashType extends Factory[HashType] {
 
+  private def intFromBytes(bytes: ByteVector): Int = {
+    require(bytes.size <= 4, "We cannot have an Int32 be larger than 4 bytes")
+    bytes.toInt(signed = false, ordering = ByteOrdering.BigEndian)
+  }
+
   def fromBytes(bytes: ByteVector): HashType = {
-    val num = Int32(bytes)
+    val num = intFromBytes(bytes)
```

Comment:
Why change from the `Int32` type to `ByteVector.toInt`?

Reply:
Int32 is in core and this was moved to crypto